### PR TITLE
[GOBBLIN-1429] Add secure TrustManager for LDAP Utils

### PR DESCRIPTION
**Pull request by abti**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1429

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
Add secure TrustManager for LDAP Utils

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-utility/src/main/java/org/apache/gobblin/util/LdapUtils.java [hunk 1]
```diff
@@ -87,7 +94,11 @@ private DirContext getDirContext(String username, String password) throws Naming
     env.put(Context.SECURITY_PRINCIPAL, username);
     env.put(Context.SECURITY_CREDENTIALS, password);
 
-    env.put("java.naming.ldap.factory.socket", TrustManagerSocketFactory.class.getCanonicalName());
+    if (_ldapUseSecureTrustManager) {
+      env.put("java.naming.ldap.factory.socket", TrustManagerSocketFactory.class.getCanonicalName());
```
Comment (treff7es) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/LdapUtils.java:98]: I think you mixed up the branches here as this one should be the secure one and the other one the non-secure one.

**Review by treff7es — CHANGES_REQUESTED** [review]

File: gobblin-utility/src/main/java/org/apache/gobblin/util/LdapUtils.java (hunk 1 above)
Comment (abti) [reply gobblin-utility/src/main/java/org/apache/gobblin/util/LdapUtils.java:98]: yes, good catch. fixing it

File: gobblin-utility/src/main/java/org/apache/gobblin/util/TrustManagerSecureSocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    this(null, null);
+  }
+
+  public TrustManagerSecureSocketFactory(String[] ciphers, String protocol) {
+    try {
+      SSLContext ctx = SSLContext.getInstance("TLS");
+      sslSocketFactory = ctx.getSocketFactory();
+      ciphers = sslSocketFactory.getSupportedCipherSuites();
+      protocol = ctx.getProtocol();
+    } catch (NoSuchAlgorithmException e) {
```
Comment (treff7es) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/TrustManagerSecureSocketFactory.java:51]: Is this intentional you silently drop this exception?

Comment (abti) [reply gobblin-utility/src/main/java/org/apache/gobblin/util/TrustManagerSecureSocketFactory.java:51]: will change this to RTE

**Review by treff7es — APPROVED** [review]

**Pull request closed without merging**